### PR TITLE
Cache subclass entity aliases in Loader

### PR DESCRIPTION
**Pull request by bahusoid**
Avoid subclass entity aliases recalculation for each row.

### CODE REVIEW

File: src/NHibernate/Loader/Loader.cs [hunk 1]
```diff
@@ -60,6 +60,11 @@ public abstract partial class Loader
 		/// </summary>
 		private bool? _canUseLimits;
 
+		/// <summary>
+		/// Caches subclass entity aliases for given persister index in <see cref="EntityPersisters"/>  and subclass entity name
+		/// </summary>
+		private readonly Dictionary<Tuple<int, string>, string[][]> _subclassEntityAliasesMap = new Dictionary<Tuple<int, string>, string[][]>();
```
Comment (fredericDelaporte) [inline src/NHibernate/Loader/Loader.cs]: This assumes a loader cannot be re-used for a different query. It seems to be true, but it is not easy to ascertain (or I have missed something).

Comment (gliljas) [reply src/NHibernate/Loader/Loader.cs]: Good point. It seems to be true, but needs more investigation.

Comment (bahusoid) [reply src/NHibernate/Loader/Loader.cs]: @fredericDelaporte I don't think it's possible.

https://github.com/nhibernate/nhibernate-core/blob/083a9280f95a2b1dfc2ecf07cc414bc064e56a1a/src/NHibernate/Loader/Loader.cs#L1692-L1696

Comment suggests that EntityAliases and other caches are calculated once. Plus there is already `_columnNameCache` cache which looks pretty query specific

Comment (bahusoid) [reply src/NHibernate/Loader/Loader.cs]: Ok. I've checked all the existing loaders - all initializations are done in ctor once. It's not really practical to reuse Loader - you need to recalculate everything. So don't know what else to add.

Again comment for `PostInstantiate` states that you *must* it call after instantiation and *will* cache select clause suffixes. Doesn't sound reusable at all.

**Pull request merged**